ui/appui-react/src/appui-react/childwindow/InternalChildWindowManager.tsx

```diff
@@ -92,8 +94,12 @@ export class InternalChildWindowManager implements FrameworkChildWindows {
     element: React.FunctionComponentElement<any>,
     container: Element | DocumentFragment
   ) {
+    // If createRoot is passed in for React 18 we have to render differently
+    // than without it. React 17 vs React 18. We need to save the root to
+    // unmount it.
     if (this._createRoot) {
-      this._createRoot(container).render(element);
+      this._root = this._createRoot(container);
+      this._root.render(element);
```

Comment:
Could we add a test for that (ideally e2e, i.e. assert on a message that is logged once a component is unmounted)?

Reply:
Added additional testing for mounting and unmounting the popout manager

Reply:
The added unit test in its current form tests React lifecycle methods, we should probably replace it with:
- Define a widget W1 that logs something on mount/unmount in our test-app
- Add a playwright test that pops out W1
- Close W1
- Assert on log messages

On the other hand, there is https://github.com/iTwin/appui/issues/524 (which if implemented would cause W1 not to unmount) so we could skip the testing for now...

Reply:
Sounds very doable, I'll add those tests

Reply:
Working on this but trying to find a way to assert the log messages with playwright, that's the hold up

Reply:
Maybe you can write a component that would `IModelApp.notifications.outputMessage(... Sticky)` on unmount, if the logger is unreachable.

Reply:
If console logs are working, maybe it will be less annoying than the sticky messages when we use the test app, depending on when your widget is displayed. (if it is only in its own frontage, that's not an issue though)

Reply:
I personally think the console.logs would be less annoying, I'll move the widget to its own fronstage

Reply:
I have it in its own fronstage and working, everything seems good. I just need to clean up some stuff. Looks like I'm running out of time today but this should be good to go Monday morning

Reply:
Added test. Put the component in the ExerciseWidgetApi frontstage